plugins/woocommerce-blocks/tests/utils/integration/editor.jsx

```diff
@@ -0,0 +1,119 @@
+/**
+ * External dependencies
+ */
+import { act, render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { useState, useEffect } from '@wordpress/element';
+import {
+	BlockEditorKeyboardShortcuts,
+	BlockEditorProvider,
+	BlockList,
+	BlockTools,
+	BlockInspector,
+	WritingFlow,
+	ObserveTyping,
+} from '@wordpress/block-editor';
+import { Popover, SlotFillProvider } from '@wordpress/components';
+import { registerCoreBlocks } from '@wordpress/block-library';
+import { ShortcutProvider } from '@wordpress/keyboard-shortcuts';
+import {
+	createBlock,
+	unregisterBlockType,
+	getBlockTypes,
+} from '@wordpress/blocks';
+import '@wordpress/format-library';
+
+/**
+ * Internal dependencies
+ */
+import { waitForStoreResolvers } from './wait-for-store-resolvers';
+
+// Polyfill for String.prototype.replaceAll until CI is runnig Node 15 or higher.
+if ( ! String.prototype.replaceAll ) {
+	String.prototype.replaceAll = function ( str, newStr ) {
+		// If a regex pattern
+		if (
+			Object.prototype.toString.call( str ).toLowerCase() ===
+			'[object regexp]'
+		) {
+			return this.replace( str, newStr );
+		}
+
+		// If a string
+		return this.replace( new RegExp( str, 'g' ), newStr );
+	};
+}
+
+/**
+ * Selects the block to be tested by the aria-label on the block wrapper, eg. "Block: Cover".
+ *
+ * @param {string} name
+ */
+export async function selectBlock( name ) {
+	await act( () => userEvent.click( screen.getByLabelText( name ) ) );
+}
+
+export function Editor( { testBlocks, settings = {} } ) {
+	const [ currentBlocks, updateBlocks ] = useState( testBlocks );
+
+	useEffect( () => {
+		return () => {
+			getBlockTypes().forEach( ( { name } ) =>
+				unregisterBlockType( name )
+			);
+		};
+	}, [] );
+
+	return (
+		<ShortcutProvider>
+			<SlotFillProvider>
+				<BlockEditorProvider
```

Comment:
This Editor function looks quite different from [one in Gutenberg](https://github.com/WordPress/gutenberg/blob/5d24debbeb3251b1b612634925bf6c0f13efcbab/test/integration/helpers/integration-test-editor.js#L64-L90). Is there any reason behind not having same structure as Gutenberg? Because it's much simpler in Gutenberg.